src/main/java/mate/academy/spring/validation/PasswordValidator.java

```diff
@@ -0,0 +1,34 @@
+package mate.academy.spring.validation;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import mate.academy.spring.dto.request.UserRequestDto;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class PasswordValidator implements ConstraintValidator<Password, UserRequestDto> {
+    private static final String PASSWORD_PATTERN =
+            "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#&()–[{}]:;',?/*~$^+=<>]).{8,20}$";
+
+    @Override
+    public boolean isValid(UserRequestDto userRequestDto,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        if (userRequestDto == null) {
+            return false;
+        }
+        String password = (String) new BeanWrapperImpl(userRequestDto)
+                .getPropertyValue("password");
+        String repeatPassword = (String) new BeanWrapperImpl(userRequestDto)
+                .getPropertyValue("repeatPassword");
+        if (password == null || repeatPassword == null) {
+            return false;
+        }
+        if (!password.equals(repeatPassword)) {
+            return false;
+        }
```

Comment:
this is too complicated and left from the improper override with `Object` type.
Also, we should keep up with the single responsibility principle and not add a size validation to this class, so let's add a separate annotation over the `password` field in the DTO.
Null check here is necessary to not get NPE when comparing 2 strings
```suggestion
        if (userRequestDto.getPassword() == null) {
            return false;
        }
```